framework/src/media/MediaWorker.cpp

```diff
@@ -59,9 +59,30 @@ void MediaWorker::stopWorker()
 	}
 }
 
-MediaQueue& MediaWorker::getQueue()
+std::function<void()> MediaWorker::deQueue()
 {
-	return mWorkerQueue;
+	return mWorkerQueue.deQueue();
+}
+
+bool MediaWorker::processLoop()
+{
+	return false;
+}
+
+int MediaWorker::mediaLooper()
+{
+	medvdbg("MediaWorker : mediaLooper\n");
+
+	while (mIsRunning) {
+		while (processLoop() && mWorkerQueue.isEmpty());
```

Comment:
I think this is the best.
With this logic, it seems that there is no need to modify queue logic.
I'll not remain TODO